influxdb3_load_generator/analysis/app.py

```diff
@@ -0,0 +1,111 @@
+    test_names = [name for name in os.listdir(RESULTS_DIRECTORY) if os.path.isdir(os.path.join(RESULTS_DIRECTORY, name))]
+    return jsonify(test_names)
+
+@app.route('/api/config-names')
+def get_config_names():
+    test_name = request.args.get('test_name')
+    test_path = os.path.join(RESULTS_DIRECTORY, test_name)
+    config_names = {}
+
+    for config_name in os.listdir(test_path):
+        config_path = os.path.join(test_path, config_name)
+        if os.path.isdir(config_path):
+            run_times = set()
+            for file_name in os.listdir(config_path):
+                match = re.search(r'_(\d{4}-\d{2}-\d{2}-\d{2}-\d{2})', file_name)
+                if match:
+                    run_time = match.group(1)
+                    run_times.add(run_time)
+            config_names[config_name] = sorted(run_times)
+
+    return jsonify(config_names)
+
+@app.route('/api/aggregated-data')
+def get_aggregated_data():
+    test_name = request.args.get('test_name')
+    config_name = request.args.get('config_name')
+    run_time = request.args.get('run_time')
+
+    config_path = os.path.join(RESULTS_DIRECTORY, test_name, config_name)
+    write_file = os.path.join(config_path, f'write_{run_time}.csv')
+    query_file = os.path.join(config_path, f'query_{run_time}.csv')
+    system_file = os.path.join(config_path, f'system_{run_time}.csv')
+
+    if os.path.isfile(write_file) and os.path.isfile(query_file) and os.path.isfile(system_file):
+        write_data = aggregate_data(write_file, 'lines', 'latency_ms')
+        query_data = aggregate_data(query_file, 'rows', 'response_ms')
+        system_data = aggregate_system_data(system_file)
+
+        aggregated_data = {
+            'config_name': config_name,
+            'run_time': run_time,
+            'write_data': write_data,
+            'query_data': query_data,
+            'system_data': system_data
+        }
+
+        return jsonify(aggregated_data)
+    else:
+        return jsonify({'error': 'Files not found for the specified configuration and run time'})
+
+def aggregate_data(file_path, lines_field, latency_field):
+    aggregated_data = []
+
+    with open(file_path, 'r') as file:
+        reader = csv.DictReader(file)
+        data = list(reader)
+
+    for row in data:
+        test_time = int(row['test_time_ms'])
+        lines = int(row[lines_field])
+        latency = int(row[latency_field])
+
+        aggregated_data.append({
+            'test_time': test_time,
+            'lines': lines,
+            'latency': latency
+        })
+
+    return aggregated_data
+
+def aggregate_system_data(file_path):
+    aggregated_data = []
+
+    with open(file_path, 'r') as file:
+        reader = csv.DictReader(file)
+        data = list(reader)
+
+    for row in data:
+        aggregated_data.append({
+            'test_time': int(row['test_time_ms']),
+            'cpu_usage': float(row['cpu_usage']),
+            'memory_bytes': int(row['memory_bytes']) / 1024 / 1024
```

Comment:
Perhaps the memory field in the reported API response should be `memory_mega_bytes`

Reply:
maybe? Seems like MB is the lowest granularity we'll care about for the entire server process.

Reply:
I meant that because the value being returned in the API response (bytes / 1024 / 1024) is not bytes anymore, but megabytes, the field name in the JSON should reflect that.

Reply:
oh haha, duh. I'll get that fixed up in a future PR 😄